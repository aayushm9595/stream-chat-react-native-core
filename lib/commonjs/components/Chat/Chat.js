var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Chat = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _dayjs = _interopRequireDefault(require("dayjs"));

var _useIsOnline2 = require("./hooks/useIsOnline");

var _useStreami18n = require("./hooks/useStreami18n");

var _ChatContext = require("../../contexts/chatContext/ChatContext");

var _ThemeContext = require("../../contexts/themeContext/ThemeContext");

var _TranslationContext = require("../../contexts/translationContext/TranslationContext");

var _package = require("../../../package.json");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Chat/Chat.tsx";

var Chat = function Chat(props) {
  var children = props.children,
      client = props.client,
      i18nInstance = props.i18nInstance,
      _props$logger = props.logger,
      logger = _props$logger === void 0 ? function () {
    return null;
  } : _props$logger,
      style = props.style;

  var _useState = (0, _react.useState)(),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      channel = _useState2[0],
      setChannel = _useState2[1];

  var _useState3 = (0, _react.useState)({
    t: function t(key) {
      return key;
    },
    tDateTimeParser: function tDateTimeParser(input) {
      return (0, _dayjs["default"])(input);
    }
  }),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      translators = _useState4[0],
      setTranslators = _useState4[1];

  (0, _useStreami18n.useStreami18n)({
    i18nInstance: i18nInstance,
    setTranslators: setTranslators
  });

  var _useIsOnline = (0, _useIsOnline2.useIsOnline)(client),
      connectionRecovering = _useIsOnline.connectionRecovering,
      isOnline = _useIsOnline.isOnline;

  (0, _react.useEffect)(function () {
    client == null ? void 0 : client.setUserAgent("stream-chat-react-native-" + _reactNative.Platform.OS + "-" + _package.version);
    client.recoverStateOnReconnect = false;
  }, []);

  var setActiveChannel = function setActiveChannel(newChannel) {
    return setChannel(newChannel);
  };

  if (!translators.t) return null;
  var chatContext = {
    channel: channel,
    client: client,
    connectionRecovering: connectionRecovering,
    isOnline: isOnline,
    logger: logger,
    setActiveChannel: setActiveChannel
  };
  return _react["default"].createElement(_ChatContext.ChatProvider, {
    value: chatContext,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 176,
      columnNumber: 5
    }
  }, _react["default"].createElement(_TranslationContext.TranslationProvider, {
    value: translators,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 177,
      columnNumber: 7
    }
  }, _react["default"].createElement(_ThemeContext.ThemeProvider, {
    style: style,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 178,
      columnNumber: 9
    }
  }, children)));
};

exports.Chat = Chat;
//# sourceMappingURL=Chat.js.map