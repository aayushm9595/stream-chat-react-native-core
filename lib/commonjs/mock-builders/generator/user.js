var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStaticUser = exports.generateUser = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateUser = function generateUser() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    banned: false,
    created_at: '2020-04-27T13:39:49.331742Z',
    id: (0, _uuid.v4)(),
    image: (0, _uuid.v4)(),
    name: (0, _uuid.v4)(),
    online: false,
    role: 'user',
    updated_at: '2020-04-27T13:39:49.332087Z'
  }, options);
};

exports.generateUser = generateUser;

var generateStaticUser = function generateStaticUser(userNumber) {
  switch (userNumber) {
    case 0:
      return generateUser({
        id: 'DannyBoi1',
        image: 'https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg',
        name: 'Dan'
      });

    case 1:
      return generateUser({
        id: 'GrantyBoi2',
        image: 'https://i.imgur.com/SLx06PP.png',
        name: 'Grant'
      });

    case 2:
      return generateUser({
        id: 'ViriBoi2',
        image: 'https://i.imgur.com/iNaC3K7.jpg',
        name: 'Vir'
      });

    default:
      return generateUser({
        id: 'NeiliBoi3',
        image: 'https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand',
        name: 'Neil'
      });
  }
};

exports.generateStaticUser = generateStaticUser;
//# sourceMappingURL=user.js.map