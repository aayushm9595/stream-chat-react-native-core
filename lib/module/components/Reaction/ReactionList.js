var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ReactionList = void 0;

var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));

var _react = _interopRequireDefault(require("react"));

var _renderReactions = require("./utils/renderReactions");

var _styledComponents = require("../../styles/styledComponents");

var _utils = require("../../utils/utils");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Reaction/ReactionList.tsx";

function _templateObject11() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-self: ", ";\n  height: 28px;\n  ", "\n  z-index: 10;\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  height: 33px;\n  width: 25px;\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  height: 33px;\n  width: 14px;\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  flex: 1;\n  height: 33px;\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  flex-direction: row;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  color: white;\n  font-size: 12px;\n  ", "\n  ", "\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  height: 33px;\n  width: 25px;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  height: 33px;\n  width: 14px;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  flex: 1;\n  height: 33px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  flex-direction: row;\n  top: -23px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-items: center;\n  flex-direction: row;\n  height: 24px;\n  padding-horizontal: 5px;\n  z-index: 1;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var leftTail = require("../../images/reactionlist/left-tail.png");

var leftCenter = require("../../images/reactionlist/left-center.png");

var leftEnd = require("../../images/reactionlist/left-end.png");

var rightTail = require("../../images/reactionlist/right-tail.png");

var rightCenter = require("../../images/reactionlist/right-center.png");

var rightEnd = require("../../images/reactionlist/right-end.png");

var Container = _styledComponents.styled.View(_templateObject(), function (_ref) {
  var theme = _ref.theme;
  return theme.message.reactionList.container.css;
});

var ImageWrapper = _styledComponents.styled.View(_templateObject2());

var LeftCenter = _styledComponents.styled.Image(_templateObject3());

var LeftEnd = _styledComponents.styled.Image(_templateObject4());

var LeftTail = _styledComponents.styled.Image(_templateObject5());

var ReactionCount = _styledComponents.styled.Text(_templateObject6(), function (_ref2) {
  var reactionCounts = _ref2.reactionCounts;
  return reactionCounts < 10 ? undefined : 'min-width: 20px;';
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.message.reactionList.reactionCount.css;
});

var Reactions = _styledComponents.styled.View(_templateObject7());

var RightCenter = _styledComponents.styled.Image(_templateObject8());

var RightEnd = _styledComponents.styled.Image(_templateObject9());

var RightTail = _styledComponents.styled.Image(_templateObject10());

var Wrapper = _styledComponents.styled.View(_templateObject11(), function (_ref4) {
  var alignment = _ref4.alignment;
  return alignment === 'left' ? 'flex-start' : 'flex-end';
}, function (_ref5) {
  var alignment = _ref5.alignment;
  return (alignment === 'left' ? 'left' : 'right') + ": -10px";
});

var ReactionList = function ReactionList(props) {
  var alignment = props.alignment,
      getTotalReactionCount = props.getTotalReactionCount,
      latestReactions = props.latestReactions,
      _props$supportedReact = props.supportedReactions,
      supportedReactions = _props$supportedReact === void 0 ? _utils.emojiData : _props$supportedReact,
      visible = props.visible;
  if (!visible) return null;
  return _react["default"].createElement(Wrapper, {
    alignment: alignment,
    testID: "reaction-list",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 175,
      columnNumber: 5
    }
  }, _react["default"].createElement(Container, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 176,
      columnNumber: 7
    }
  }, _react["default"].createElement(Reactions, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 177,
      columnNumber: 9
    }
  }, (0, _renderReactions.renderReactions)(latestReactions, supportedReactions)), _react["default"].createElement(ReactionCount, {
    reactionCounts: getTotalReactionCount(supportedReactions),
    __source: {
      fileName: _jsxFileName,
      lineNumber: 180,
      columnNumber: 9
    }
  }, getTotalReactionCount(supportedReactions))), _react["default"].createElement(ImageWrapper, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 186,
      columnNumber: 7
    }
  }, alignment === 'left' ? _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(LeftTail, {
    source: leftTail,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 189,
      columnNumber: 13
    }
  }), _react["default"].createElement(LeftCenter, {
    resizeMode: "stretch",
    source: leftCenter,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 190,
      columnNumber: 13
    }
  }), _react["default"].createElement(LeftEnd, {
    source: leftEnd,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 191,
      columnNumber: 13
    }
  })) : _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(RightEnd, {
    source: rightEnd,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 195,
      columnNumber: 13
    }
  }), _react["default"].createElement(RightCenter, {
    resizeMode: "stretch",
    source: rightCenter,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 196,
      columnNumber: 13
    }
  }), _react["default"].createElement(RightTail, {
    source: rightTail,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 197,
      columnNumber: 13
    }
  }))));
};

exports.ReactionList = ReactionList;
//# sourceMappingURL=ReactionList.js.map