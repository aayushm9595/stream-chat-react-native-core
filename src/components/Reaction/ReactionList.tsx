import React from 'react';

import { renderReactions } from './utils/renderReactions';

import { styled } from '../../styles/styledComponents';
import { emojiData } from '../../utils/utils';

import type { ImageRequireSource } from 'react-native';

import type {
  Alignment,
  MessageWithDates,
} from '../../contexts/messagesContext/MessagesContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../types/types';

const leftTail: ImageRequireSource = require('../../images/reactionlist/left-tail.png');
const leftCenter: ImageRequireSource = require('../../images/reactionlist/left-center.png');
const leftEnd: ImageRequireSource = require('../../images/reactionlist/left-end.png');

const rightTail: ImageRequireSource = require('../../images/reactionlist/right-tail.png');
const rightCenter: ImageRequireSource = require('../../images/reactionlist/right-center.png');
const rightEnd: ImageRequireSource = require('../../images/reactionlist/right-end.png');

const Container = styled.View`
  align-items: center;
  flex-direction: row;
  height: 24px;
  padding-horizontal: 5px;
  z-index: 1;
  ${({ theme }) => theme.message.reactionList.container.css}
`;

const ImageWrapper = styled.View`
  flex-direction: row;
  top: -23px;
`;

const LeftCenter = styled.Image`
  flex: 1;
  height: 33px;
`;

const LeftEnd = styled.Image`
  height: 33px;
  width: 14px;
`;

const LeftTail = styled.Image`
  height: 33px;
  width: 25px;
`;

const ReactionCount = styled.Text<{ reactionCounts: number }>`
  color: white;
  font-size: 12px;
  ${({ reactionCounts }) =>
    reactionCounts < 10 ? undefined : 'min-width: 20px;'}
  ${({ theme }) => theme.message.reactionList.reactionCount.css}
`;

const Reactions = styled.View`
  flex-direction: row;
`;

const RightCenter = styled.Image`
  flex: 1;
  height: 33px;
`;

const RightEnd = styled.Image`
  height: 33px;
  width: 14px;
`;

const RightTail = styled.Image`
  height: 33px;
  width: 25px;
`;

const Wrapper = styled.View<{ alignment: Alignment }>`
  align-self: ${({ alignment }) =>
    alignment === 'left' ? 'flex-start' : 'flex-end'};
  height: 28px;
  ${({ alignment }) => `${alignment === 'left' ? 'left' : 'right'}: -10px`}
  z-index: 10;
`;

export type LatestReactions<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = MessageWithDates<At, Ch, Co, Me, Re, Us>['latest_reactions'];

export type Reaction = {
  icon: string;
  id: string;
};

export type ReactionListProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  alignment: Alignment;
  getTotalReactionCount: (
    arg: {
      icon: string;
      id: string;
    }[],
  ) => number;
  latestReactions: LatestReactions<At, Ch, Co, Me, Re, Us>;
  /**
   * e.g.,
   * [
   *  {
   *    id: 'like',
   *    icon: 'üëç',
   *  },
   *  {
   *    id: 'love',
   *    icon: '‚ù§Ô∏èÔ∏è',
   *  },
   *  {
   *    id: 'haha',
   *    icon: 'üòÇ',
   *  },
   *  {
   *    id: 'wow',
   *    icon: 'üòÆ',
   *  },
   * ]
   */
  supportedReactions: Reaction[];
  visible: boolean;
};

/**
 * @example ./ReactionList.md
 */
export const ReactionList = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  props: ReactionListProps<At, Ch, Co, Me, Re, Us>,
) => {
  const {
    alignment,
    getTotalReactionCount,
    latestReactions,
    supportedReactions = emojiData,
    visible,
  } = props;

  if (!visible) return null;

  return (
    <Wrapper alignment={alignment} testID='reaction-list'>
      <Container>
        <Reactions>
          {renderReactions(latestReactions, supportedReactions)}
        </Reactions>
        <ReactionCount
          reactionCounts={getTotalReactionCount(supportedReactions)}
        >
          {getTotalReactionCount(supportedReactions)}
        </ReactionCount>
      </Container>
      <ImageWrapper>
        {alignment === 'left' ? (
          <>
            <LeftTail source={leftTail} />
            <LeftCenter resizeMode='stretch' source={leftCenter} />
            <LeftEnd source={leftEnd} />
          </>
        ) : (
          <>
            <RightEnd source={rightEnd} />
            <RightCenter resizeMode='stretch' source={rightCenter} />
            <RightTail source={rightTail} />
          </>
        )}
      </ImageWrapper>
    </Wrapper>
  );
};
