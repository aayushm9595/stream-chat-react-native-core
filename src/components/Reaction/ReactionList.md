```js
import { View } from 'react-native';

import { ReactionList } from './ReactionList';

const props = {
  alignment: 'right',
  getTotalReactionCount: () => 5,
  latestReactions: [
    {
      created_at: '2019-11-28T11:14:20.155817Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'sad',
      updated_at: '2019-11-28T11:14:20.155817Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-28T11:13:54.845187Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'haha',
      updated_at: '2019-11-28T11:13:54.845187Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-28T10:32:58.981307Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'like',
      updated_at: '2019-11-28T10:32:58.981307Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-26T22:32:21.975407Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'sad',
      updated_at: '2019-11-26T22:32:21.975407Z',
      user: {
        banned: false,
        created_at: '2019-04-02T13:26:55.274436Z',
        id: 'billowing-firefly-8',
        image:
          'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
        last_active: '2020-01-06T16:47:25.692477Z',
        name: 'Billowing firefly',
        online: true,
        role: 'user',
        updated_at: '2020-01-06T16:47:25.696327Z',
      },
      user_id: 'billowing-firefly-8',
    },
    {
      created_at: '2019-11-26T22:31:51.31847Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'like',
      updated_at: '2019-11-26T22:31:51.31847Z',
      user: {
        banned: false,
        created_at: '2019-04-02T13:26:55.274436Z',
        id: 'billowing-firefly-8',
        image:
          'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
        last_active: '2020-01-06T16:47:25.692477Z',
        name: 'Billowing firefly',
        online: true,
        role: 'user',
        updated_at: '2020-01-06T16:47:25.696327Z',
      },
      user_id: 'billowing-firefly-8',
    },
  ],
  visible: true,
};
<View style={{ width: 90 }}>
  <ReactionList {...props} />
</View>;
```
