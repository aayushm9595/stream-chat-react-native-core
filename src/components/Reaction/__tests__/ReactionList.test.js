import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import { generateUser } from 'mock-builders/generator/user';
import { getTestClientWithUser } from 'mock-builders/mock';

import { ReactionList } from '../ReactionList';

import { Chat } from '../../Chat/Chat';

const defaultProps = {
  getTotalReactionCount: () => 5,
  latestReactions: [
    {
      created_at: '2019-11-28T11:14:20.155817Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'sad',
      updated_at: '2019-11-28T11:14:20.155817Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-28T11:13:54.845187Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'haha',
      updated_at: '2019-11-28T11:13:54.845187Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-28T10:32:58.981307Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'like',
      updated_at: '2019-11-28T10:32:58.981307Z',
      user: {
        banned: false,
        created_at: '2019-08-26T10:49:19.667963Z',
        id: 'bender',
        image: 'https://bit.ly/321RmWb',
        last_active: '2020-01-06T08:42:44.097139Z',
        name: 'Bender',
        online: false,
        role: 'user',
        updated_at: '2020-01-06T08:42:44.106042Z',
      },
      user_id: 'bender',
    },
    {
      created_at: '2019-11-26T22:32:21.975407Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'sad',
      updated_at: '2019-11-26T22:32:21.975407Z',
      user: {
        banned: false,
        created_at: '2019-04-02T13:26:55.274436Z',
        id: 'billowing-firefly-8',
        image:
          'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
        last_active: '2020-01-06T16:47:25.692477Z',
        name: 'Billowing firefly',
        online: true,
        role: 'user',
        updated_at: '2020-01-06T16:47:25.696327Z',
      },
      user_id: 'billowing-firefly-8',
    },
    {
      created_at: '2019-11-26T22:31:51.31847Z',
      message_id: 'billowing-firefly-8-52540509-ed51-476e-8b6b-e8929aba9e33',
      score: 1,
      type: 'like',
      updated_at: '2019-11-26T22:31:51.31847Z',
      user: {
        banned: false,
        created_at: '2019-04-02T13:26:55.274436Z',
        id: 'billowing-firefly-8',
        image:
          'https://stepupandlive.files.wordpress.com/2014/09/3d-animated-frog-image.jpg',
        last_active: '2020-01-06T16:47:25.692477Z',
        name: 'Billowing firefly',
        online: true,
        role: 'user',
        updated_at: '2020-01-06T16:47:25.696327Z',
      },
      user_id: 'billowing-firefly-8',
    },
  ],
  visible: true,
};

const emojiTypes = ['sad', 'haha', 'like'];

describe('ReactionList', () => {
  const clientUser = generateUser();
  let chatClient;

  const getComponent = (props = {}) => (
    <Chat client={chatClient}>
      <ReactionList {...props} />
    </Chat>
  );

  beforeEach(async () => {
    chatClient = await getTestClientWithUser(clientUser);
  });

  it('should render left aligned ReactionList', async () => {
    const { queryByTestId, toJSON } = render(
      getComponent({ ...defaultProps, alignment: 'left' }),
    );

    await waitFor(() => {
      expect(queryByTestId('reaction-list')).toBeTruthy();
      emojiTypes.forEach((emoji) => expect(queryByTestId(emoji)).toBeTruthy());
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render right aligned ReactionList', async () => {
    const { queryByTestId, toJSON } = render(
      getComponent({ ...defaultProps, alignment: 'right' }),
    );

    await waitFor(() => {
      expect(queryByTestId('reaction-list')).toBeTruthy();
      emojiTypes.forEach((emoji) => expect(queryByTestId(emoji)).toBeTruthy());
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render not visible ReactionList', async () => {
    const { queryByTestId, toJSON } = render(
      getComponent({ ...defaultProps, visible: false }),
    );

    await waitFor(() => {
      expect(queryByTestId('reaction-list')).toBeFalsy();
      emojiTypes.forEach((emoji) => expect(queryByTestId(emoji)).toBeFalsy());
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
