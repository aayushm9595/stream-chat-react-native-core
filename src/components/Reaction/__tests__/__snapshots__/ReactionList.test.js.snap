// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionList should render left aligned ReactionList 1`] = `
<View
  alignment="left"
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "height": 28,
        "left": -10,
        "zIndex": 10,
      },
    ]
  }
  testID="reaction-list"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 24,
          "paddingHorizontal": 5,
          "zIndex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        testID="sad"
      >
        😔
      </Text>
      <Text
        testID="haha"
      >
        😂
      </Text>
      <Text
        testID="like"
      >
        👍
      </Text>
    </View>
    <Text
      reactionCounts={5}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 12,
          },
        ]
      }
    >
      5
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "top": -23,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/left-tail.png",
        }
      }
      style={
        Array [
          Object {
            "height": 33,
            "width": 25,
          },
        ]
      }
    />
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/left-center.png",
        }
      }
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 33,
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/left-end.png",
        }
      }
      style={
        Array [
          Object {
            "height": 33,
            "width": 14,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ReactionList should render not visible ReactionList 1`] = `null`;

exports[`ReactionList should render right aligned ReactionList 1`] = `
<View
  alignment="right"
  style={
    Array [
      Object {
        "alignSelf": "flex-end",
        "height": 28,
        "right": -10,
        "zIndex": 10,
      },
    ]
  }
  testID="reaction-list"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 24,
          "paddingHorizontal": 5,
          "zIndex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        testID="sad"
      >
        😔
      </Text>
      <Text
        testID="haha"
      >
        😂
      </Text>
      <Text
        testID="like"
      >
        👍
      </Text>
    </View>
    <Text
      reactionCounts={5}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 12,
          },
        ]
      }
    >
      5
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "top": -23,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/right-end.png",
        }
      }
      style={
        Array [
          Object {
            "height": 33,
            "width": 14,
          },
        ]
      }
    />
    <Image
      resizeMode="stretch"
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/right-center.png",
        }
      }
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 33,
          },
        ]
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../src/images/reactionlist/right-tail.png",
        }
      }
      style={
        Array [
          Object {
            "height": 33,
            "width": 25,
          },
        ]
      }
    />
  </View>
</View>
`;
